app/src/main/java/com/example/criminalintentapp/presentation/authentication/AuthenticationViewModel.kt

```diff
@@ -2,35 +2,35 @@ package com.example.criminalintentapp.presentation.authentication
 
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
 import com.google.firebase.auth.FirebaseAuth
 import com.google.firebase.auth.FirebaseUser
+import kotlinx.coroutines.launch
 
-class AuthenticationViewModel : ViewModel() {
+class AuthenticationViewModel(val service: FirebaseAuthService) : ViewModel() {
```

Comment:
Would be nice to have this renamed `service` -> `authService` so it is more intuitive on what it actually does. Also, consider changing it to a private val